hermes-common/src/main/java/pl/allegro/tech/hermes/infrastructure/schema/repo/JerseySchemaRepoClient.java

```diff
@@ -28,7 +42,24 @@ public boolean isSubjectRegistered(String subject) {
 
     @Override
     public void registerSchema(String subject, String schema) {
-        target.path(subject).path("register").request().put(Entity.entity(schema, MediaType.TEXT_PLAIN));
+        Response response = target.path(subject).path("register").request().put(Entity.entity(schema, MediaType.TEXT_PLAIN));
+        checkResponse(response.getStatusInfo(), subject, response.readEntity(String.class));
+    }
+
+    private void checkResponse(Response.StatusType statusType, String subject, String response) {
+        switch (statusType.getFamily()) {
+            case SUCCESSFUL:
+                logger.info("Successful write to schema repo for subject {}", subject);
+                break;
+            case CLIENT_ERROR:
+                logger.warn("Invalid schema for subject {}. Details: {}", subject, response);
+                throw new InvalidSchemaException("Invalid schema. For more details see schema-repo logs.");
+            case SERVER_ERROR:
+                logger.error("Failure write to schema repo for subject {}. Reason: {}", subject, response);
+                throw new SchemaRepoException("Failure write to schema-repo.");
+            default:
+                break;
```

Comment:
IMO, just in case, it's better to throw IllegalStateException here than to pass for this unexpected condition silently.